### PR TITLE
[PT-5597] (4) `Catalog::construct_order_parameters` testing coverage

### PR DESCRIPTION
**Pull request by andher1802**
See changelog.

Checklist:
* [x] Test coverage
* [x] Version bump
* [x] Changelog update

### CODE REVIEW

File: tests/test_catalog.py [hunk 1]
```diff
@@ -27,6 +28,8 @@
 }
 POINT_BBOX = (1.0, 2.0, 1.0, 2.0)
 
+Geometry = catalog.Geometry
```
Comment (andher1802) [inline tests/test_catalog.py:31]: This one is due to a pylance issue, which does not allow to use imported variables in the test parameters typings.

File: tests/test_catalog.py [hunk 2]
```diff
@@ -113,7 +114,7 @@ def test_should_track_order_status(
 
 class TestCatalog:
     host = "oneatlas"
-    catalog = catalog.Catalog(auth=mock.MagicMock(), workspace_id=constants.WORKSPACE_ID)
+    catalog_obj = catalog.Catalog(auth=mock.MagicMock(), workspace_id=constants.WORKSPACE_ID)
```
Comment (javidq) [inline tests/test_catalog.py:117]: Not needed, self.catalog doesn't shadow package name

File: up42/catalog.py [hunk 3]
```diff
@@ -274,14 +274,7 @@ def construct_order_parameters(
...
         image_id: str,
-        aoi: Union[
-            dict,
-            geojson.Feature,
-            geojson.FeatureCollection,
-            list,
-            geopandas.GeoDataFrame,
-            geom.Polygon,
-        ] = None,
+        aoi: Optional[Geometry] = None,
```
Comment (javidq) [inline up42/catalog.py:277]: Let's simplify the schema handling via initializing the params in advance

File: tests/test_catalog.py [hunk 4]
```diff
@@ -349,13 +350,57 @@ def test_should_estimate_order(self, requests_mock: req_mock.Mocker, auth_mock:
...
+        )
+        order_parameters: order.OrderParams = catalog.Catalog(
+            auth_mock, constants.WORKSPACE_ID
+        ).construct_order_parameters(
+            data_product_id=constants.DATA_PRODUCT_ID,
+            image_id=image_id,
+            aoi=aoi,
+            tags=tags,
+        )
+        assert schema_property in order_parameters["params"]
```
Comment (javidq) [inline tests/test_catalog.py:400]: To be simplified to a single equality check via
```python
expected = {
            "params": {
                "id": image_id,
                schema_property: None,
                **({"aoi": aoi} if aoi else {}),
            },
            "dataProduct": constants.DATA_PRODUCT_ID,
            **({"tags": tags} if tags else {}),
        }
```

File: tests/test_catalog.py [hunk 5]
```diff
@@ -349,13 +350,57 @@ def test_should_estimate_order(self, requests_mock: req_mock.Mocker, auth_mock:
...
-    assert list(order_parameters.keys()) == ["dataProduct", "params"]
-    assert order_parameters["params"]["acquisitionMode"] is None
+    def test_should_construct_order_parameters(
+        self,
+        auth_mock: mock.MagicMock,
+        requests_mock: req_mock.Mocker,
+        aoi: Optional[catalog.Geometry],
+        tags: Optional[List[str]],
+    ):
+        schema_property = "any-property"
```
Comment (javidq) [inline tests/test_catalog.py:382]: schema_property -> required_property

File: tests/test_catalog.py [hunk 6]
```diff
@@ -349,13 +350,57 @@ def test_should_estimate_order(self, requests_mock: req_mock.Mocker, auth_mock:
...
+        image_id = str(uuid.uuid4())
+        url_schema = f"{constants.API_HOST}/orders/schema/{constants.DATA_PRODUCT_ID}"
+        requests_mock.get(
+            url_schema,
+            json={
+                "required": [schema_property],
+                "properties": {schema_property: {"type": "string", "title": "string", "format": "string"}},
+            },
+        )
+        order_parameters: order.OrderParams = catalog.Catalog(
```
Comment (javidq) [inline tests/test_catalog.py:392]: typing hint is not needed here

File: CHANGELOG.md [hunk 7]
```diff
@@ -30,6 +30,11 @@ You can check your current version with the following command:
 
 For more information, see [UP42 Python package description](https://pypi.org/project/up42-py/).
 
+## 2.1.0a9
```
Comment (javidq) [inline CHANGELOG.md:33]: We don't need fixture_catalog any more

**Review by javidq — CHANGES_REQUESTED** [review]

**Review by javidq — APPROVED** [review]

**Pull request merged**